File: 5 Periodo/Padrões/final-project/Factory Method/factoryMethod-typescript/fabricas/FabricaFord.ts
import { Carro } from '../carros/Carro'
import { Fiesta } from '../carros/Fiesta'
import { FabricaDeCarro } from './FabricaDeCarro'

export class FabricaFord extends FabricaDeCarro {

	public criarCarro(): Carro {
		return new Fiesta();
	}

}

File: 5 Periodo/Padrões/final-project/Factory Method/factoryMethod-java/src/br/edu/ifpb/padroes/diegofrazao/fabricas/FabricaFord.java
package br.edu.ifpb.padroes.diegofrazao.fabricas;

import br.edu.ifpb.padroes.diegofrazao.carros.Carro;
import br.edu.ifpb.padroes.diegofrazao.carros.Fiesta;

public class FabricaFord implements FabricaDeCarro {
    @Override
    public Carro criarCarro() {
        return new Fiesta();
    }
}
